streams/src/test/java/org/apache/kafka/test/MockInternalNewProcessorContext.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.test;
+
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.processor.MockProcessorContext;
```

Comment:
This is the one defined in `test-utils`, right? Should we be using the new PAPI one (https://github.com/apache/kafka/blob/trunk/streams/test-utils/src/main/java/org/apache/kafka/streams/processor/api/MockProcessorContext.java) instead?

Reply:
Good catch. One of the things I found migrating this mock was that timestamp is not defined. 
At the moment I just added a new `setTimestamp` method, but seems rare to set timestamp for the context and set the timestamp for the record.

Will check if setting the timestamp when forwarding be a better approach.

Reply:
> Will check if setting the timestamp when forwarding be a better approach.

It's not. Forward is called "too late" to check it in this suppress test cases. Staying with `setTimestamp` could be good enough then.

Reply:
Sounds good. I think we can afford to cross that bridge later on as well.